modules/apps/analytics/analytics-client-js/src/analytics.js

```diff
@@ -126,16 +126,24 @@ class Analytics {
 		ENV.Analytics.create = Analytics.create;
 		ENV.Analytics.dispose = Analytics.dispose;
 
+		let email = '';
+		let name = '';
+
 		if (
 			Liferay &&
 			Liferay.ThemeDisplay &&
-			Liferay.ThemeDisplay.getUserEmailAddress()
+			Liferay.ThemeDisplay.getUserEmailAddress &&
+			Liferay.ThemeDisplay.getUserName
 		) {
-			self.setIdentity({
-				email: Liferay.ThemeDisplay.getUserEmailAddress(),
-			});
+			email = Liferay.ThemeDisplay.getUserEmailAddress();
+			name = Liferay.ThemeDisplay.getUserName();
 		}
 
+		self.setIdentity({
+			email,
+			name,
+		});
```

Comment:
Can't you run set Identity inside condition? This way we don't need to create variables

Reply:
If we do that then `setIdentity` won't run if the `Liferay` object doesn't exist.